### PR TITLE
Rt685 flash remove code location

### PR DESCRIPTION
**Pull request by EmilioCBen**
Removed the CODE_LOCATION choice as part of the flexspi clock init setup and 
replaced the usage with the Kconfig FLASH_MCUX_FLEXSPI_XIP instead.

### CODE REVIEW

**Comment by carlescufi** [comment]
@dleach02 @danieldegrasse @mmahadevan108 please review

File: soc/arm/nxp_imx/Kconfig [hunk 1]
```diff
@@ -23,4 +23,14 @@ config SOC_PART_NUMBER
...
 	default SOC_PART_NUMBER_IMX8MQ_M4 if SOC_SERIES_IMX8MQ_M4
 
+config FLASH_MCUX_FLEXSPI_XIP
+	bool "MCUX FlexSPI flash access with xip"
+	default y
+	depends on MEMC_MCUX_FLEXSPI
+	depends on (CODE_FLEXSPI || CODE_FLEXSPI2 || SOC_SERIES_IMX_RT6XX)
+	select XIP
+	help
+	  Allows using the flash API while running in XIP.
```
Comment (danieldegrasse) [inline soc/arm/nxp_imx/Kconfig:33]: We need to update the help string for this Kconfig, as it is no longer is specific to the flash driver

**Review by dleach02 — DISMISSED** [review]

File: soc/arm/nxp_imx/Kconfig [hunk 2]
```diff
@@ -23,4 +23,15 @@ config SOC_PART_NUMBER
 	default SOC_PART_NUMBER_IMX8ML_M7 if SOC_SERIES_IMX8ML_M7
 	default SOC_PART_NUMBER_IMX8MQ_M4 if SOC_SERIES_IMX8MQ_M4
 
+config FLASH_MCUX_FLEXSPI_XIP
+	bool "MCUX FlexSPI flash access with xip"
+	default $(dt_node_has_compat, $(DT_CHOSEN_FLASH_PARENT), $(DT_COMPAT_FLEXSPI))
```
Comment (danieldegrasse) [inline soc/arm/nxp_imx/Kconfig:28]: We still need this KConfig to be enabled for the RT1xxx series. I have not verified that the `zephyr,flash` chosen node is the 
SPI flash chip for all RT boards. If this KConfig isn't set, flash samples will likely fail, as the flash driver will attempt to reconfigure the FlexSPI while running in XIP, which will likely crash the application. Have you tested this change with a flash sample on a R6xx, RT5xx, and RT1xxx board?

Comment (EmilioCBen) [reply soc/arm/nxp_imx/Kconfig:28]: I have not tested these SOC's yet, I can test real quick with the RT595

**Review by danieldegrasse — APPROVED** [review]

**Review by carlescufi — APPROVED** [review]

**Pull request merged**